### PR TITLE
[CarthageKit/Archive.swift] Adopt API Design Guidelines

### PR DESCRIPTION
**Pull request by ikesyo**
#1558

### CODE REVIEW

File: Source/CarthageKit/Archive.swift [hunk 1]
```diff
@@ -13,7 +13,7 @@ import ReactiveTask
 
 /// Zips the given input items (recursively) into an archive that will be
 /// located at the given URL.
-public func zipIntoArchive(destinationArchiveURL: NSURL, workingDirectory: String, inputPaths: [String]) -> SignalProducer<(), CarthageError> {
+public func zipArchive(into destinationArchiveURL: NSURL, workingDirectory: String, inputPaths: [String]) -> SignalProducer<(), CarthageError> {
```
Comment (mdiep) [inline Source/CarthageKit/Archive.swift]: This doesn't seem quite right. It's not zipping the archive itself; it's zipping into it.

Maybe we should rearrange things like so?

```
zip(paths: [String], into archiveURL: NSURL, workingDirectory: String)
```

File: Source/CarthageKit/Archive.swift [hunk 2]
```diff
@@ -26,7 +26,7 @@ public func zipIntoArchive(destinationArchiveURL: NSURL, workingDirectory: Strin
 
 /// Unzips the archive at the given file URL, extracting into the given
 /// directory URL (which must already exist).
-public func unzipArchiveToDirectory(fileURL: NSURL, _ destinationDirectoryURL: NSURL) -> SignalProducer<(), CarthageError> {
+public func unzipArchive(_ fileURL: NSURL, to destinationDirectoryURL: NSURL) -> SignalProducer<(), CarthageError> {
```
Comment (mdiep) [inline Source/CarthageKit/Archive.swift]: Maybe move _archive_ to the parameter?

```
unzip(archive fileURL: NSURL, to destinationURL: NSURL)
```

File: Source/CarthageKit/Archive.swift [hunk 3]
```diff
@@ -38,7 +38,7 @@ public func unzipArchiveToDirectory(fileURL: NSURL, _ destinationDirectoryURL: N
 
 /// Unzips the archive at the given file URL into a temporary directory, then
 /// sends the file URL to that directory.
-public func unzipArchiveToTemporaryDirectory(fileURL: NSURL) -> SignalProducer<NSURL, CarthageError> {
+public func unzipArchiveToTemporaryDirectory(_ fileURL: NSURL) -> SignalProducer<NSURL, CarthageError> {
```
Comment (mdiep) [inline Source/CarthageKit/Archive.swift]: Maybe make this just `unzip(archive fileURL: NSURL)` to match the above?

**Comment by ikesyo** [comment]
Thanks for the feedback! I've addressed them.

**Pull request merged**